vNext/UnityGLTFv2.cs

```diff
@@ -1,63 +1,70 @@
+namespace UnityGLTF {
+
 /// <summary>
 /// Handles importing object from schema into Unity and handles exporting of objects from Unity into schema
 /// </summary>
-public partial class UnityGLTFImporter
+public virtual class Importer
 {
-    public UnityGLTFImporter(
+    public Importer(
         IDataLoader dataLoader,
-        UnityGLTFImporterConfig config = new UnityGLTFImporterConfig()
+        ImporterConfig config = new ImporterConfig()
         );
 
     /// <summary>
     /// Creates a Unity GameObject from a glTF scene
     /// </summary>
-    /// <param name="unityGLTFFObject">Object which contains information to parse</param>
+    /// <param name="unityGLTFObject">Object which contains information to parse</param>
     /// <param name="sceneId">Scene of the glTF to load</param>
+    /// <param name="progress">Progress of load completion</param>
     /// <returns>The created Unity object</returns>
-    public Task<GameObject> ImportSceneAsync(
-        UnityGLTFObject unityGLTFFObject,
-        int sceneId = -1
+    public virtual Task<GameObject> ImportSceneAsync(
+        UnityGLTFObject unityGLTFObject,

```

Comment:
The whole UnityGLTFObject stateful wrapper still seems like a confusing part of this design to me. I think I'm now leaning more towards just passing in the raw deserialized data structure for the gltf root, and additionally (optionally) passing in an AssetCache by ref. Any problems with that approach?

Reply:
It doesn't enforce that asset caches and paired to a glTF object. It's just decoupling the two rather than having them stored in an object. I should update the design to have UnityGLTFObject expose the AssetCache.